### DISCUSSION TITLE
Deconstructing HAL objects

### DISCUSSION THREAD

**Discussion by xPhilippeRoux** [HALRobotics/Beta#91] [Ideas]
Do you plan on adding a possibility to deconstruct usual HAL objects ?

Ex: PART > alias + geometry + frame

Thanks.

Philippe

**Comment by sebandraos** [comment]
Hi @XtreeE-PRO ,
This is something that we have access to internally for debugging but they tend to rely heavily on knowing how the Framework works behind the scenes with some additional complexities linked to how the Grasshopper graph runs. e.g. If you create a Part and immediately get its data, that data won't be updated by the Grasshopper component during the execution of a Procedure so you end up with potentially false information.
How do you envisage using this sort of component? That will help us to try and find a useful solution rather than just throwing tens of outputs at you and hoping you can work out what they all mean.
Also, which objects would you like to be able to decompose?

**Reply by xPhilippeRoux** [reply]
Hi @sebandraos ,
Thanks for that. I'm transferring specific lists of data between blocks in my canvas. I would be happy to use your HAL objects, not only because it's more clean and clear, but also because it enables to transfer only a small number of elements and use its meta-data to do the actual calculations.
Parts are the best example for me right now, but I might want to use other in the future.

**Reply by sebandraos** [reply]
Ok, so you're using the Parts as a proxy to group meshes, frames etc. but only ever in static configurations. This is definitely possible and we will try and incorporate something like this into an update soon

**Reply by sebandraos** [reply]
@XtreeE-PRO For parts, you should be able to cast them into Planes or Meshes by using the GH parameters as shown below. Hopefully that's what you were after.
![image](https://user-images.githubusercontent.com/6084391/114526766-30c4ab00-9c3f-11eb-916e-37e454564085.png)

**Reply by xPhilippeRoux** [reply]
@sebandraos This is exactly what I'm looking for, thanks for it. Do you think however that it will be stable with future updates, as it is an implicit deconstruction ?

**Reply by sebandraos** [reply]
As we're in beta nothing is 100% stable and things are liable to change but it is actually exactly because this is an implicit conversion that these are very unlikely to change as there is nothing else to do with them.
